examples/godocbot/pkg/controller/pullrequest/godoc_deployer.go

```diff
@@ -0,0 +1,265 @@
+)
+
+// GodocDeployer watches PullRequest object which have a commitID specified in
+// their Spec and deploys a Godoc deployment which runs godoc server for the PR.
+// It watches the PullRequest object for changes in commitID and reconciles the
+// generated godoc deployment.
+type GodocDeployer struct {
+	controller.Controller
+}
+
+func NewGodocDeployer(mgr manager.Manager) (*GodocDeployer, error) {
+	prReconciler := &pullRequestReconciler{
+		Client: mgr.GetClient(),
+	}
+
+	// Setup a new controller to Reconcile PullRequests
+	c, err := controller.New("pull-request-controller", mgr, controller.Options{Reconcile: prReconciler})
+	if err != nil {
+		return nil, err
+	}
+
+	// Watch PullRequest objects
+	err = c.Watch(
+		&source.Kind{Type: &v1alpha1.PullRequest{}},
+		&handler.Enqueue{})
+	if err != nil {
+		return nil, err
+	}
+
+	// Watch deployments generated for PullRequests objects
+	err = c.Watch(
+		&source.Kind{Type: &appsv1.Deployment{}},
+		&handler.EnqueueOwner{
+			OwnerType:    &v1alpha1.PullRequest{},
+			IsController: true,
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return &GodocDeployer{Controller: c}, nil
+}
+
+// pullRequestReconciler ensures there is a godoc deployment is running with
+// the commitID specified in the PullRequest object.
+type pullRequestReconciler struct {
+	Client client.Client
+}
+
+func (r *pullRequestReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	ctx := context.Background()
+
+	log.Printf("reconciling request '%v'", request.NamespacedName)
+	// Fetch PullRequest object
+	pr := &v1alpha1.PullRequest{}
+	err := r.Client.Get(ctx, request.NamespacedName, pr)
+	if errors.IsNotFound(err) {
+		log.Printf("Could not find PullRequest %v.\n", request)
+		return reconcile.Result{}, nil
+	}
+
+	if err != nil {
+		log.Printf("Could not fetch PullRequest %v for %+v\n", err, request)
+		return reconcile.Result{}, err
+	}
+
+	if pr.Spec.CommitID == "" {
+		log.Printf("Waiting for PR %s commitID to be updated", request.NamespacedName)
+		return reconcile.Result{}, nil
+	}
+
+	dp := &appsv1.Deployment{}
+	err = r.Client.Get(ctx, request.NamespacedName, dp)
+	if errors.IsNotFound(err) {
+		log.Printf("Could not find deployment for PullRequest %v. creating deployment", request)
+
+		dp, err = deploymentForPullRequest(pr)
+		if err != nil {
+			log.Printf("error creating new deployment for PullRequest: %v %v", request, err)
+			return reconcile.Result{}, nil
+		}
+		if err = r.Client.Create(ctx, dp); err != nil {
+			log.Printf("error creating new deployment for PullRequest: %v %v", request, err)
+			return reconcile.Result{}, err
+		}
+		log.Printf("created deployment for %v successfully", request.NamespacedName)
+	}
+
+	prinfo, _ := parsePullRequestURL(pr.Spec.URL)
+	prinfo.commitID = pr.Spec.CommitID
+
+	if len(dp.Spec.Template.Spec.Containers[0].Args) <= 5 || (pr.Spec.CommitID != dp.Spec.Template.Spec.Containers[0].Args[5]) {
```

Comment:
Maybe there is a better way of doing this.  Patching the args to a cmd is actually pretty common.